lib/ja_serializer/builder/relationship.ex

```diff
@@ -17,10 +17,14 @@ defmodule JaSerializer.Builder.Relationship do
   end
 
   defp add_links(relation, {_type, _name, opts}, context) do
-    case opts[:link] do
-      nil ->  relation
-      path -> Map.put(relation, :links, [Link.build(context, :related, path)])
-    end
+    Keyword.get(opts, :relationship_links, [])
+      |> Keyword.put_new(:related, opts[:link])
+      |> Enum.filter(fn {_key, value} -> is_binary(value) end)
```

Comment:
I was doing that to weed out a nil that can creep in from putting in the `link` key on 21.  I've been giving this some though and I wonder if it would be better to move that logic up into the area where the other opts are normalized ?